python/lsst/ts/salobj/configurable_csc.py

```diff
@@ -0,0 +1,253 @@
+# This file is part of ts_salobj.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ConfigurableCsc"]
+
+import abc
+import pathlib
+import subprocess
+import types
+
+import yaml
+
+from . import base
+from .base_csc import BaseCsc, State
+from .validator import DefaultingValidator
+
+# maximum length of settingsVersion.recommendedSettingsVersion field
+MAX_LABELS_LEN = 256
+
+
+class ConfigurableCsc(BaseCsc, abc.ABC):
+    """Base class for a configurable Commandable SAL Component (CSC)
+
+    Parameters
+    ----------
+    sallib : ``module``
+        salpy component library generatedby SAL
+    index : `int` or `None`
+        SAL component index, or 0 or None if the component is not indexed.
+    config_dir : `str`
+        Path to configuration files.
```

Comment:
Should a config_directory be able to be changed, or is it set once and then thats it.

Reply:
It is set when the CSC is constructed. Normally the `config_dir` constructor argument will be `None` and `ConfigurableCsc` will work out where to find the correct directory (see the last comment on this ticket for more information). But that code was not in place when you commented on this pull request.